clients/vault/src/metrics.rs

```diff
@@ -1,27 +1,522 @@
+	pub static ref COLLATERALIZATION: GaugeVec =
+		GaugeVec::new(Opts::new("collateralization", "Collateralization"), &[CURRENCY_LABEL])
+			.expect("Failed to create prometheus metric");
+	pub static ref REQUIRED_COLLATERAL: GaugeVec =
+		GaugeVec::new(Opts::new("required_collateral", "Required Collateral"), &[CURRENCY_LABEL])
+			.expect("Failed to create prometheus metric");
+	pub static ref MEAN_IDLE_DURATION: IntGaugeVec =
+		IntGaugeVec::new(Opts::new("mean_idle_duration_ms", "Total Idle Duration"), &[TASK_NAME])
+			.expect("Failed to create prometheus metric");
+	pub static ref MEAN_POLL_DURATION: IntGaugeVec =
+		IntGaugeVec::new(Opts::new("mean_poll_duration_ms", "Total Poll Duration"), &[TASK_NAME])
+			.expect("Failed to create prometheus metric");
+	pub static ref MEAN_SCHEDULED_DURATION: IntGaugeVec = IntGaugeVec::new(
+		Opts::new("mean_scheduled_duration_ms", "Total Scheduled Duration"),
+		&[TASK_NAME]
+	)
+	.expect("Failed to create prometheus metric");
+	pub static ref XLM_BALANCE: GaugeVec = GaugeVec::new(
+		Opts::new("stellar_balance", "Stellar Balance"),
+		&[CURRENCY_LABEL, XLM_BALANCE_TYPE_LABEL]
+	)
+	.expect("Failed to create prometheus metric");
+	pub static ref ISSUES: GaugeVec = GaugeVec::new(
+		Opts::new("issue_count", "Number of issues"),
+		&[CURRENCY_LABEL, REQUEST_STATUS_LABEL]
+	)
+	.expect("Failed to create prometheus metric");
+	pub static ref REDEEMS: GaugeVec = GaugeVec::new(
+		Opts::new("redeem_count", "Number of redeems"),
+		&[CURRENCY_LABEL, REQUEST_STATUS_LABEL]
+	)
+	.expect("Failed to create prometheus metric");
+	pub static ref NATIVE_CURRENCY_BALANCE: Gauge =
+		Gauge::new("native_currency_balance", "Native Currency Balance")
+			.expect("Failed to create prometheus metric");
+	pub static ref FEE_BUDGET_SURPLUS: GaugeVec =
+		GaugeVec::new(Opts::new("fee_budget_surplus", "Fee Budget Surplus"), &[CURRENCY_LABEL])
+			.expect("Failed to create prometheus metric");
 	pub static ref RESTART_COUNT: IntCounter =
 		IntCounter::new("restart_count", "Number of service restarts")
 			.expect("Failed to create prometheus metric");
 }
+const STELLAR_NATIVE_ASSET_TYPE: [u8; 6] = [110, 97, 116, 105, 118, 101]; //"native"
+
+#[derive(Clone, Debug)]
+struct XLMBalance {
+	upperbound: Gauge,
+	lowerbound: Gauge,
+	actual: Gauge,
+}
+
+#[derive(Clone, Debug)]
+struct RequestCounter {
+	open_count: Gauge,
+	completed_count: Gauge,
+	expired_count: Gauge,
+}
+
+#[derive(Clone, Debug)]
+pub struct PerCurrencyMetrics {
+	locked_collateral: Gauge,
+	collateralization: Gauge,
+	required_collateral: Gauge,
+	asset_balance: XLMBalance,
+	issues: RequestCounter,
+	redeems: RequestCounter,
+}
+
+#[async_trait]
+pub trait VaultDataReader {
+	async fn get_entries(&self) -> Vec<VaultData>;
+}
+
+#[async_trait]
+impl VaultDataReader for VaultIdManager {
+	async fn get_entries(&self) -> Vec<VaultData> {
+		self.get_entries().await
+	}
+}
+
+pub const fn one() -> f64 {
+	10u128.pow(12u32) as f64
+}
```

Comment:
I think this function is quite useless and only used at the top. Maybe we could use the `one()` method that exists on our assets on the `CurrencyId` enum already.

Reply:
[It does exist.](https://github.com/pendulum-chain/spacewalk/blob/c81caf406ce63b7dffbe8443fcff6e29e77ca96d/primitives/src/lib.rs#L478) I added it recently.

Reply:
Oh. Got it. Thanks. Did not see it previous time! Will use your code!

Reply:
Done.